builder/remotecontext/git/gitutils.go

```diff
@@ -70,7 +70,7 @@ func fetchArgs(remoteURL *url.URL, ref string) []string {
 	shallow := true
 
 	if strings.HasPrefix(remoteURL.Scheme, "http") {
-		res, err := http.Head(fmt.Sprintf("%s/info/refs?service=git-upload-pack", remoteURL))
+		res, err := http.Get(fmt.Sprintf("%s/info/refs?service=git-upload-pack", remoteURL))
```

Comment:
I think we should avoid special casing any particular provider whenever we can so that a) all providers can benefit from improvements (be they performance, robustness, protocol compatibility) and b) we avoid building in reliance on the behaviour's of particular services as they happen to be configured today and miss out on future improvements etc.